server/notifications.go

```diff
@@ -99,13 +117,46 @@ func (ah *ActivityHandler) handleCreatedActivityNotification(msg *clientmodels.M
 			post.FileIds,
 			skippedFileAttachments,
 		)
+		if err != nil {
+			ah.plugin.GetAPI().LogWarn(
+				"Failed to deliver notification for chat member away from Teams",
+				"error", err,
+				"user_id", mattermostUserID,
+				"teams_user_id", member.UserID,
+				"chat_id", chat.ID,
+				"message_id", msg.ID,
+				"activity", presences[member.UserID].Activity,
+				"availability", presences[member.UserID].Availability,
+			)
+			ah.plugin.metricsService.ObserveNotification(isGroupChat, hasFiles, metrics.DiscardedReasonInternalError)
+			continue
+		}
+
+		ah.plugin.GetAPI().LogInfo(
+			"Delivered notification for chat member away from Teams",
+			"user_id", mattermostUserID,
+			"teams_user_id", member.UserID,
+			"chat_id", chat.ID,
+			"message_id", msg.ID,
+			"activity", presences[member.UserID].Activity,
+			"availability", presences[member.UserID].Availability,
```

Comment:
Seeing fields needing to be defined for every log message like this really makes me appreciate how loggers like logrus allow you to slowly build a custom logger for each flow that contains important field values. Would we ever consider making important field keys constants and/or making field helper functions like userIDField(mattermostUserID) to help eliminate typos or allow for renaming a field key? Not sure I even think that would be a good idea, but figured I would float it by you.

Reply:
> Seeing fields needing to be defined for every log message like this really makes me appreciate how loggers like logrus allow you to slowly build a custom logger for each flow that contains important field values. 

Yes, I'm in full alignment. I "stuck to the basics" for this PR, but got fed up trying to replicate the fields over at https://github.com/mattermost/mattermost-plugin-msteams/pull/749/files and ended up just importing logrus there.

> Would we ever consider making important field keys constants and/or making field helper functions...

Oh wow, yes! I spent a bunch of clean up early on in this project fixing `user_id` and `userID` to make grepping logs more useful. Formalizing a `withUser([string|model.User])` would be even better!

I've added this idea to the pool over at https://mattermost.atlassian.net/wiki/spaces/Toolkit/pages/3044966406/Toolkit+Improvements+Q4+2024.